Hybrid/example.rb

```diff
@@ -0,0 +1,270 @@
+  puts 'Press any key to continue'
+  gets
+  puts 'Turning off the virtual machine...'
+  compute_client.virtual_machines.power_off(GROUP_NAME, vm.name)
+
+  puts 'Your virtual machine is now off. Lets start the virtual machine.'
+  puts 'Press any key to continue'
+  gets
+  puts 'Starting the virtual machine...'
+  compute_client.virtual_machines.start(GROUP_NAME, vm.name)
+
+  puts 'Your virtual machine has started. Lets restarting the virtual machine.'
+  puts 'Press any key to continue'
+  gets
+  puts 'Re-Starting the virtual machine...'
+  compute_client.virtual_machines.restart(GROUP_NAME, vm.name)
+  puts 'Your virtual machine is now on.'
+
+  puts 'Press any key to continue and delete the sample resources'
+  gets
+
+  # Delete Resource group and everything in it
+  puts 'Delete Resource Group'
+  resource_client.resource_groups.delete(GROUP_NAME)
+  puts "\nDeleted: #{GROUP_NAME}"
+
+end
+
+def print_group(resource)
+  puts "\tname: #{resource.name}"
+  puts "\tid: #{resource.id}"
+  puts "\tlocation: #{resource.location}"
+  puts "\ttags: #{resource.tags}"
+  puts "\tproperties:"
+  print_item(resource.properties)
+end
+
+def print_item(resource)
+  resource.instance_variables.sort.each do |ivar|
+    str = ivar.to_s.gsub /^@/, ''
+    if resource.respond_to? str.to_sym
+      puts "\t\t#{str}: #{resource.send(str.to_sym)}"
+    end
+  end
+  puts "\n\n"
+end
+
+def export_template(resource_client)
+  puts "Exporting the resource group template for #{GROUP_NAME}"
+  export_result = resource_client.resource_groups.export_template(
+      GROUP_NAME,
+      ResourceModels::ExportTemplateRequest.new.tap{ |req| req.resources = ['*'] }
+  )
+  puts export_result.template
+  puts ''
+end
+
+# Create a Virtual Machine and return it
+def create_vm(compute_client, network_client, location, vm_name, storage_acct, subnet, public_ip)
+  puts "Creating a network interface for the VM #{vm_name}"
+  print_item nic = network_client.network_interfaces.create_or_update(
+      GROUP_NAME,
+      "sample-ruby-nic-#{vm_name}",
+      NetworkModels::NetworkInterface.new.tap do |interface|
+        interface.location = LOCAL
+        interface.ip_configurations = [
+            NetworkModels::NetworkInterfaceIPConfiguration.new.tap do |nic_conf|
+              nic_conf.name = "sample-ruby-nic-#{vm_name}"
+              nic_conf.private_ipallocation_method = NetworkModels::IPAllocationMethod::Dynamic
+              nic_conf.subnet = subnet
+              nic_conf.public_ipaddress = public_ip
+            end
+        ]
+      end
+  )
+
+  puts 'Creating a Ubuntu 16.04.0-LTS Standard DS2 V2 virtual machine w/ a public IP'
+  vm_create_params = ComputeModels::VirtualMachine.new.tap do |vm|
+    vm.location = location
+    vm.os_profile = ComputeModels::OSProfile.new.tap do |os_profile|
+      os_profile.computer_name = vm_name
+      os_profile.admin_username = 'notAdmin'
+      os_profile.admin_password = 'Pa$$w0rd92'
```

Comment:
Is there a NewGuid method instead of harcoding a password? We can print the password.. low pri to fix